src/components/forms/CreateNhsAppTemplate/server-action.ts

```diff
@@ -1,82 +1,75 @@
 import { TemplateFormState } from '@utils/types';
-import { zodValidationServerAction } from '@utils/zod-validation-server-action';
 import { z } from 'zod';
-import { saveSession } from '@utils/form-actions';
+import { saveTemplate } from '@utils/form-actions';
+import { redirect, RedirectType } from 'next/navigation';
 
-const schema = z.object({
-  formId: z.enum(['create-nhs-app-template-back', 'create-nhs-app-template'], {
-    message: 'Internal server error',
-  }),
+const $CreateNhsAppTemplateSchema = z.object({
+  nhsAppTemplateName: z
+    .string({ message: 'Enter a template name' })
```

Comment:
Do we need the message here? we didn't have it before.

Reply:
it ensures you'll still get a coherent error message if nhsAppTemplateName is missing rather than an empty string